sru/1851858/esm-trusty-test-2.md

```diff
@@ -0,0 +1,107 @@
+
+2.
+ a. Start with a fresh Ubuntu **non-x86** instance which does have u-a-t installed. Enable esm with 'ubuntu-advantage attach'. Upgrade to u-a-t from -proposed.
+ b. Confirm that machine remains attached, esm gets **disabled**, and status remains **disabled**.
+ c. Update machine token files to represent delta in architectures from contract server. Ensure `ua refresh` retains attached, but esm status becomes **n/a**
+
+```
+export UA_CONTRACT_TOKEN=<NewContractToken>
+export ARCHIVE_URL=http://archive.ubuntu.com/ubuntu
+export ADMIN_NET_ID=
+export SSH_KEY=
+export LP_USER=
+
+echo -- BEGIN test 2a: enable esm via `ubuntu-advantage attach <TOKEN>` on typical trusty-updates cloud-images which already have -updates installed
+
+# Launch a basic trusty cloud-image that is updated to latest ubuntu-advantage-tools from -updates
+cat > update-uat-trusty.yaml <<EOF
+#cloud-config
+ssh_import_id: [$LP_USER]
+package_update: true
+package_upgrade: true
+runcmd:
+ - apt-get install -qy ubuntu-advantage-tools 
+EOF
+
+echo "Creating image list for ppc64el"
+openstack image list --property architecture=ppc64el -f json > images.json
+
+# create a floating ip, keep the IP and floating-ip UUID for cleanup
+FIP=( `openstack floating ip create ext_net -f json | jq -r '. | .floating_ip_address + " " + .id'` )
+VM_IP=${FIP[0]}; FIP_ID=${FIP[1]}
+
+
+sshopts=( -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR )
+
+cat > ppa-key << EOF
+-----BEGIN PGP PUBLIC KEY BLOCK-----
+
+xo0EUs00cgEEAJJqaPue5gzQiLB1krT9slYbqVW/bSBpW9+qX8gFI44IVM/Bo3yh
+9BPAs1RAzja96N0FS6SNlew4JYfk7MBT2sFDGpm3bTKt9Go7muO0JkvKv0vYgrrw
+qORlWK3SfsYa6EpsCdVzZPAKvGzc8I0XywVgcJhM5okx+3J2naBaSp9NABEBAAHN
+K0xhdW5jaHBhZCBQUEEgZm9yIENJIFRyYWluIFBQQSBTZXJ2aWNlIFRlYW3CuAQT
+AQIAIgUCUs00cgIbAwYLCQgHAwIGFQgCCQoLBBYCAwECHgECF4AACgkQhVBBKOzx
+IEy62gP/T2h98ongV+RXekM1DpgTNoH0PBHrZVj4zfrvrYKZOaxRmJ6TWtzG8tFI
+uB4gPjaFeenJBhCFaZ9UncFQemS9jztQ/pA049L1N7Tijd8/BKD7gc7tM07+Fq+Q
+6DT7VuUFiVlfZUwWYzk5UXEk6ctluoIRpnRWUHmh6NssuAgd1Nk=
+=aPbC
+-----END PGP PUBLIC KEY BLOCK-----
+EOF
+
+cat > add_uat_apt_pocket.sh << EOF
+#/bin/bash
+pocket_name=\$1
+if [ "\$pocket_name" = "devel" ]; then
+  echo deb [trusted=yes] http://ppa.launchpad.net/ci-train-ppa-service/3830/ubuntu trusty main | tee /etc/apt/sources.list.d/\$pocket_name.list
+  apt-key add /ppa-key
+else
+  echo deb $ARCHIVE_URL \$pocket_name main | tee /etc/apt/sources.list.d/\$pocket_name.list
+fi
+EOF
+
+for series in trusty; do
+  echo "### BEGIN $series"
+  # Determine image, launch instance and attach IP address
+  image=$(cat images.json | jq -r '.[]|select((.Name|contains("disk1.img")) and (.Name|contains("'$series'"))) | .ID' | tail -n 1)
+  openstack server create --flavor m1.small --image $image --key-name $SSH_KEY --nic net-id=$ADMIN_NET_ID --user-data sethostname.yaml test-$series --wait
```

Comment:
what is `sethostname.yaml`?

Reply:
Why do we need `--nic net-id`? What should I set it to?

Reply:
For serverstack I had to attach a nic and floating IP to access the machine (probably because I didn't properly set ingress security routes for ssh to the private IP). It's unnneded in canonistack as long as I add the right ingress.